flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectResultFetcher.java

```diff
@@ -67,20 +66,23 @@ public class CollectResultFetcher<T> {
     public CollectResultFetcher(
             AbstractCollectResultBuffer<T> buffer,
             CompletableFuture<OperatorID> operatorIdFuture,
-            String accumulatorName) {
-        this(buffer, operatorIdFuture, accumulatorName, DEFAULT_RETRY_MILLIS);
+            String accumulatorName,
+            long resultFetchTimeout) {
+        this(buffer, operatorIdFuture, accumulatorName, DEFAULT_RETRY_MILLIS, resultFetchTimeout);
```

Comment:
Should we move the DEFAULT_RETRY_MILLIS to the CollectResultIterator?
In this way, we can only keep one constructor for CollectResultFetcher. In the constructor of CollectResultIterator, we can use this(xxx) to avoid different logic.

Reply:
I'm lean to leave it in the `CollectResultFetcher` because I consider it as an internal mechanism of `CollectResultFetcher`.